woocommerce/woocommerce
# Add link into the error snackbar

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:






This PR adds a link to the error messages. This PR is part of the changes described in https://github.com/woocommerce/woocommerce/issues/46768

![Screenshot 2024-06-10 at 10 49 19 AM](https://github.com/woocommerce/woocommerce/assets/1314156/63f0c47e-e4ec-41d2-8c12-610e7cc8759b)

### ACs
- [ ] If there is a field with an error in a non-visible tab, the error snackbar won't dismiss automatically and it will have a link to the error and a close button (`x`).
- [ ] Clicking the error link should navigate to the tab containing the field with an error and focus on the field. Currently, this works for: product name, regular price, and sale price.
- [ ] If there is a field with an error in a visible tab, the error snackbar won't include a close button (`x`) and will dismiss automatically. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure `Try the new product editor (Beta)` is enabled at `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add New.
3. Add a regular price (but do not add a product name), and go to the Inventory tab.
4. Click Publish.
5. A snackbar with the message: `Product name is required. View error` should appear.
6. Press `View error`. You should be redirected to the General tab with the name field focused.
7. Repeat the test with a negative regular price or a negative sale price. The behavior should be the same.
8. Go to the Variations tab and create a few variations.
9. Select a single variation, navigate to the Shipping tab and toggle the `This variation requires shipping or pickup`.
10. Press `Update`.
11. A snackbar with the message: `Regular price is required. View error` should appear.
12. Press `View error`. You should be redirected to the General tab with the regular price field focused.
13. Add a price and switch to the Inventory tab.
14. Add a duplicated SKU.
15. Return to the General tab and click Update.
16. A snackbar with the message: `Invalid or duplicated SKU. View error` should appear.
17. Press `View error`. You should be redirected to the Inventory tab. Note that this error type is not fully handled yet.
18. Same with the dimensions checks under the Shipping tab. This will be fixed after merging [this PR](https://github.com/woocommerce/woocommerce/pull/49357).



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
